File: server/app/Models/Store.cs
using System.ComponentModel.DataAnnotations;

namespace app.Models;

public class Store{
	public Guid Id{ get; init; } = Guid.CreateVersion7();
	[StringLength(50)] public required string Name{ get; set; }
	[StringLength(50)] public required string Address{ get; set; }
	[StringLength(50)] public required string City{ get; set; }
	[StringLength(13)] public required string ContactNumber{ get; set; }
	public required TimeOnly OpeningTime{ get; set; }
	public required TimeOnly ClosingTime{ get; set; }
	public required bool IsWarehouse{ get; set; }
}

File: client/src/models/Store.ts
export default class Store {
    id: string;
    name: string;
    address: string;
    city: string;
    contactNumber: string;
    openingHours: string;
    closingHours: string;
    isWarehouse: boolean;
    
    constructor(jsonData: any) {
        ({
            id: this.id,
            name: this.name,
            address: this.address,
            city: this.city,
            contactNumber: this.contactNumber,
            openingTime: this.openingHours,
            closingTime: this.closingHours,
            isWarehouse: this.isWarehouse,
        } = jsonData);

    }
}